### ISSUE TITLE
End-to-End Experiments

### ISSUE THREAD

**Issue by stefan-grafberger** [stefan-grafberger/mlwhatif#27]
# Description
* We want to perform several end-to-end experiments for mlwhatif.
* We want to test four scenarios: 
    * `DataCorruption` on the test set only
    * `PermutationFeatureImportance` (see [here](https://scikit-learn.org/stable/modules/permutation_importance.html) and [here](https://christophm.github.io/interpretable-ml-book/feature-importance.html)), which also only requires tet side changes
    * `DataCleaning` on both train and test side
    * `OperatorImpact` on both train and test side
* For this, we want to use five different datasets:
    * Sneakers (For this, only `DataCorruption` and `DataCleaning` for mislabels only are applicable)
    * Reviews (4 joins, ca. 100,000 rows)
    * Cardio (currently has 0 joins, will need to insert 2 artificial joins, 70,000 rows)
    * Folkstables (0 joins, choose something with around 30,000 rows)
    * Healthcare (replace the existing join with an artificial fuzzy-merge join)
* Then, we should have 5 pipelines inspired by dspipes (ca. 3) and openml (ca. 2), that range from having cheap to expensive featurization
    * E.g., for text `HashingVectorizer`, then `CountVectorizer` + `TfidfTransformer`, finally a `Word2Vec` transformer. Similarly for categorical and numerical columns
* Test two different data access costs: normal pandas `read_csv` and also more expensive https access, for this I can put a copy of the data into a public GitHub repository, pandas also allows data reading via https
* We want to test three different models
    * Cheap: `LogisticRegression`
    * Medium: `Xgboost`
    * Expensive: `SVC `
    * For Sneakers, we will use a CNN only
* We need to make sure that we can measure runtime of all relevant execution steps: end-to-end, instrumented pipeline only, estimated what-if runtime, actual what-if runtime, multi-query-optimization, etc.
